examples/profiling/index.js

```diff
@@ -0,0 +1,38 @@
+const { start, stop } = require('@splunk/otel');
+const { diag, DiagConsoleLogger, DiagLogLevel, trace, SpanStatusCode, context } = require('@opentelemetry/api');
+
+// If OTEL_LOG_LEVEL env var is set, configure logger
+if (process.env.OTEL_LOG_LEVEL) {
+  diag.setLogger(new DiagConsoleLogger(), DiagLogLevel[process.env.OTEL_LOG_LEVEL]);
+}
+
+// Profiling is still experimental and has to be enabled explicitly
+start({
+	profiling: {
+		callstackInterval: 100,
+		collectionDuration: 1_000,
+	},
+});
+
+const doWork = () => {
+	const start = Date.now();
+	while (Date.now() - start < 2000) {}
+};
+
+// setTimeout has to be here because profiling is currently started asyncronously to avoid blocking the runtime.
+// If we didn't we'd run stop before the profiling has started in the background.
```

Comment:
Could make the native init run in the same loop step, don't think the `setImmediate` is important there. Can do it separately tho

Reply:
Will change that separately from adding the example.